### PR TITLE
[wk-libs] use wk test data for tests

### PR DESCRIPTION
**Pull request by jstriebel**
### Description:
- uses the e2e testdata for the tests where possible, adapts the tests accordingly
- uses the e2e user token for the tests
- ensures that local setup snapshots and docker snapshots are (almost) the same
- allows the download of public annotations and datasets from wk-instances that are not in the current token-context

### Issues:
- fixes most things from #514

### CODE REVIEW

File: webknossos/test.sh [hunk 1]
```diff
@@ -1,28 +1,40 @@
 #!/usr/bin/env bash
 set -eEuo pipefail
 
+export WK_TOKEN=1b88db86331a38c21a0b235794b9e459856490d70408bcffb767f64ade0f83d2bdb4c4e181b9a9a30cdece7cb7c65208cc43b6c1bb5987f5ece00d348b1a905502a266f8fc64f0371cd6559393d72e031d0c2d0cabad58cccf957bb258bc86f05b5dc3d4fff3d5e3d9c0389a6027d861a21e78e3222fb6c5b7944520ef21761e
+export WK_URL=http://localhost:9000
+export DOCKER_TAG=master__16748
```
Comment (fm3) [inline webknossos/test.sh:6]: add print about the used wk tag, NOTE that this may have to be updated etc etc

File: webknossos/test.sh [hunk 2]
```diff
@@ -1,28 +1,40 @@
...
+
 if [ $# -eq 1 ] && [ "$1" = "--refresh-snapshots" ]; then
     if ! curl -sf localhost:9000/api/health; then
```
Comment (fm3) [inline webknossos/test.sh:9]: please add print if tests run against already-running localhost:9000 or not

File: webknossos/tests/test_generated_client.py [hunk 3]
```diff
@@ -98,7 +98,9 @@ def test_user_list(auth_client: Client) -> None:
 
 
 @pytest.mark.vcr()
-def test_dataset_info(client: Client) -> None:
+def test_dataset_info() -> None:
+    with webknossos_context():
```
Comment (fm3) [inline webknossos/tests/test_generated_client.py:102]: naming is ambiguous, change defaults to fallbacks

File: webknossos/webknossos/annotation/annotation.py [hunk 4]
```diff
@@ -126,21 +126,31 @@ def load(cls, annotation_path: Union[str, PathLike]) -> "Annotation":
...
-            + "please adapt it e.g. via 'with webknossos_context(…)'"
-        )
+
         from webknossos.client._download_annotation import download_annotation
 
-        return download_annotation(annotation_type, annotation_id)
+        if webknossos_url != _get_context().url:
+            warnings.warn(
+                f"The supplied url {webknossos_url} does not match your current context {_get_context().url}. "
+                + "Only public annotations can be downloaded this way, using no token. "
```
Comment (fm3) [inline webknossos/webknossos/annotation/annotation.py:143]:
```suggestion
                + "Without a token, only public annotations can be downloaded."
```

File: webknossos/webknossos/dataset/dataset.py [hunk 5]
```diff
@@ -214,12 +225,27 @@ def download(
...
         from webknossos.client._download_dataset import download_dataset
+        from webknossos.client.context import _get_context, webknossos_context
 
-        return download_dataset(
-            dataset_name, organization_name, bbox, layers, mags, path, exist_ok
-        )
+        if webknossos_url is not None and webknossos_url != _get_context().url:
+            warnings.warn(
+                f"The supplied url {webknossos_url} does not match your current context {_get_context().url}. "
+                + "Only public datasets can be downloaded this way, using no token. "
```
Comment (fm3) [inline webknossos/webknossos/dataset/dataset.py:236]: grammar see above

**Review by fm3 — APPROVED** [review]
LGTM apart from my small nitpicks :)

File: webknossos/tests/test_generated_client.py (hunk 3 above)
Comment (jstriebel) [reply webknossos/tests/test_generated_client.py:102]: I'll adapt this in #577

**Pull request merged**